### DISCUSSION TITLE
[categorie_corrigee] Simplification de l'algorithme Cas 1 & 2 & 3

### DISCUSSION THREAD

**Discussion by JMPivette** [JMPivette/evavelo#5] [Ideas]
Les algorithmes défini pour les **Cas 1, 2 & 3** dans le chapitre 3.1.11 peuvent être simplifié comme suit:

- S’il y a des réponses aux questions spécifiques des itinérants(Q25-27), le cycliste sera attribué à la catégorie **"Itinérant"**.
- Si le type de sortie déclarée (Q2) est « une randonnée sur plusieurs jours » ET la durée de séjour est supérieure à 1 jour (Q18), le cycliste sera attribué à la catégorie **"Itinérant"**
- Dans tous les autres cas, le cycliste sera attribué à la catégorie **[categorie]** (réponse au questionnaire)

**Comment by JMPivette** [comment]
Extrait de la méthodologie:

![image](https://user-images.githubusercontent.com/46813298/103012473-f6be8480-453b-11eb-98bb-afe2c928e2ce.png)

**Comment by JMPivette** [comment]
Algorithme Cas 2 et Cas 3 sont identiques à algorithme Cas 1.

**Reply by SteffMangin** [reply]
Je ne suis pas totalement persuadée par la simplification que tu proposes. Il faudrait que je prenne le temps d'écrire les cas pour vérifier mais il me semble qu'il y a un pb avec les histoires : 
Si le type de sortie déclarée est « une randonnée sur plusieurs jours » et que la durée de séjour est inférieure ou égale à 1 jour, alors :
o S’il y a des réponses aux questions spécifiques des itinérants, le cycliste sera attribué à la catégorie « itinérant ».
o S’il n’y a pas de réponse aux questions spécifiques des itinérants, le cycliste sera attribué à la catégorie « xxx».

Je dois sortir une bonne partie de la matinée. Est-ce que si je regarde ce midi cela te convient ?

**Reply by JMPivette** [reply]
Pas d'urgence, je ne reprendrai que lundi.

Pour y voir plus clair, j'ai représenté sous forme de Diagramme de Venn.

Le cas de figure que tu mentionnes est bien inclus dans ma simplification:
- Question spécifique itinérant -> "itinérant" dans tous les cas.
- Pas de question spécifique + "Plusieurs journées" + dms <= 1 -> catégorie xxx: cela rentre bien dans les "autres cas" de ma simplification.

**Reply by SteffMangin** [reply]
Salut, j'espère que tu vas bien et que le père Noël a été généreux. 
Tes propositions semblent en effet fonctionner comme il faut. Je dois être tordue où alors c'est la pure transcription d'une méthode d'analyse en appliquant des filtres sous Excel... 
C'est donc OK pour moi. A voir à la lumière des tests si quelque chose ne nous a pas échappé.
